site/sfguides/src/intro_to_snowpark_container_services/intro_to_snowpark_container_services.md

```diff
@@ -0,0 +1,544 @@
+```sql
+USE ROLE SPCS_USER_ROLE;
+USE DATABASE SPCS_HOL_DB;
+USE SCHEMA PUBLIC;
+USE WAREHOUSE SPCS_HOL_WH;
+
+CREATE OR REPLACE TABLE WEATHER (
+    DATE DATE,
+    LOCATION VARCHAR,
+    TEMP_C NUMBER,
+    TEMP_F NUMBER
+);
+
+INSERT INTO weather (DATE, LOCATION, TEMP_C, TEMP_F) 
+    VALUES 
+        ('2023-03-21', 'London', 15, NULL),
+        ('2023-07-13', 'Manchester', 20, NULL),
+        ('2023-05-09', 'Liverpool', 17, NULL),
+        ('2023-09-17', 'Cambridge', 19, NULL),
+        ('2023-11-02', 'Oxford', 13, NULL),
+        ('2023-01-25', 'Birmingham', 11, NULL),
+        ('2023-08-30', 'Newcastle', 21, NULL),
+        ('2023-06-15', 'Bristol', 16, NULL),
+        ('2023-04-07', 'Leeds', 18, NULL),
+        ('2023-10-23', 'Southampton', 12, NULL);
+```
+Now, let's create a function that specifies our `convert-api` service's `convert-api` endpoint:
+```sql
+CREATE OR REPLACE FUNCTION convert_udf (input float)
+RETURNS float
+SERVICE=CONVERT_API      //Snowpark Container Service name
+ENDPOINT='convert-api'   //The endpoint within the container
+MAX_BATCH_ROWS=5         //limit the size of the batch
+AS '/convert';           //The API endpoint
+```
+We can now test our function:
+```sql
+SELECT convert_udf(12) as conversion_result;
+```
+And even update our table to populate the `TEMP_F` field using our Service Function:
+```sql
+UPDATE WEATHER
+SET TEMP_F = convert_udf(TEMP_C);
+
+SELECT * FROM WEATHER;
+```
+
+<!-- ------------------------ -->
+## Managing Services with SQL
+Duration: 5
+
+There are a number of useful functions we should explore with respect to controlling the service itself from SQL. More information on SQL commands can be found at [Snowpark Container Services: Working with Services](https://docs.snowflake.com/en/LIMITEDACCESS/snowpark-containers/working-with-services)
+
+1. Get the status of your container using CALL $SYSTEM:
+
+    From a SQl console:
+
+    ```sql
+    CALL SYSTEM$GET_SERVICE_STATUS('<database_name>.PUBLIC.JUPYTER_SNOWPARK_SERVICE');
+    ```
+
+2. Check the status of the logs with :
+
+    From a SQl console:
+
+    ```sql
+    CALL SYSTEM$GET_SERVICE_LOGS('<database_name>.PUBLIC.JUPYTER_SNOWPARK_SERVICE', '0', 'jupyter-snowpark',10);
+    ```
+
+3. Suspend your container using the ALTER SERVICE command
+
+    From a SQL console:
+
+    ```sql
+    ALTER SERVICE <database_name>.PUBLIC.JUPYTER_SNOWPARK_SERVICE SUSPEND ;
+    ```
+
+<!-- ------------------------ -->
+## Stop the Services and Suspend the Compute Pool
+Duration: 2
+
+If you no longer need the services and compute pool up and running, we can stop the services and suspend the compute pool so that we don't incur any cost (Snowpark Container Services bill credits/second based on the compute pool's uptime, similar to Virtual Warehouse billing):
+```sql
+USE ROLE SPCS_USER_ROLE;
+ALTER COMPUTE POOL SPCS_HOL_POOL STOP ALL;
+ALTER COMPUTE POOL SPCS_HOL_POOL SUSPEND;
+```
+
```

Comment:
Do you want to do all the other cleanup?
- Drop the roles, drop the database, drop the warehouse, etc